### ISSUE TITLE
Reduce 1_notmnist.ipynb memory requirements.

### ISSUE THREAD

**Issue by vincentvanhoucke** [tensorflow/tensorflow#936]
Lots of people are having trouble downloading and pickling the data. need to look into ways to make that easier.

**Comment by bhack** [comment]
Please Vincent add also an intermediate pickle of the dataset. The first cell where train and test is formed is slow to run every time you break your notebook session.

**Comment by vincentvanhoucke** [comment]
@bhack under which conditions do you 'lose' the pickle? Is it because you're using an ephemeral docker container?

**Comment by bhack** [comment]
I don't lose the pickle but I think it is better to save also one at an intermediate point. What happens if people doesn't run the pickle step? You have safe condition  for not run again the download but some cell are also heavy so it is safer to save train and test set as soon as formed.

**Comment by vincentvanhoucke** [comment]
@rthouvenin your approach is a net improvement over the status quo.
Would you be ok cleaning it up (in a way that can be easily diffed) and sending me a PR?
https://github.com/rthouvenin/tensorflow/blob/notmnist-lowmem/tensorflow/examples/udacity/1_notmnist.ipynb

**Comment by rthouvenin** [comment]
Sure, just opened the PR (I got what you meant by easily diffed, notebook doesn't play nicely on that aspect... :)).

Though I did not have trouble to download the files (it just took time), I was also a bit scared it would be difficult as my connection is quite erratic sometimes. I don't think there is much to be done, but I would suggest warning the reader about it, and suggest to download the file separately if they have a slow / not reliable connection (e.g. with a download manager that support pause/resume). 
To help with that, maybe a torrent link could be set up (I didn't check if there is already one).
